### PR TITLE
protocols/relay/examples: Add client server instructions

### PR DESCRIPTION
**Pull request by mxinden**
Add instructions on how to run a three node example where a listening
relay client listens via a relay server and a dialing relay client dials
the listening relay client via the relay server.

//CC @r-zig

### CODE REVIEW

**Review by r-zig — CHANGES_REQUESTED** [review]
I think that the 3 client (the dial one) command should be with client-dial argument instead of client-listen and add /p2p/ before the listening client peer id:
`cargo run --example relay -- client-**dial**<addr-relay-server>/p2p/<peer-id-relay-server>/p2p-circuit/**p2p/**<peer-id-listening-relay-client>

File: protocols/relay/examples/relay.rs [hunk 1]
```diff
@@ -18,14 +18,41 @@
...
+//! 1. To start the relay server, run `cargo run --example relay -- relay`.
+//!
+//! 2. To start the listening relay client run `cargo run --example relay -- client-listen
+//! <addr-relay-server>/p2p/<peer-id-relay-server>/p2p-circuit` in a second terminal where:
+//!
+//!   - `<addr-relay-server>` is replaced by one of the listening addresses of the relay server.
+//!   - `<peer-id-relay-server>` is replaced by the peer id of the relay server.
+//!
+//! 3. To start the dialing relay client run `cargo run --example relay -- client-listen
+//! <addr-relay-server>/p2p/<peer-id-relay-server>/p2p-circuit/<peer-id-listening-relay-client>` in
```
Comment (r-zig) [inline protocols/relay/examples/relay.rs:36]: suppose that the relay peer id is: QmRelay and the relay peer address is /ip6/::1/tcp/54443
the listening peer id is:  QmListenPeer

cargo run --example relay -- client-dial /ip6/::1/tcp/54443/p2p/QmRelay/p2p-circuit/p2p/QmListenPeer

**Review by r-zig — CHANGES_REQUESTED** [review]

**Comment by mxinden** [comment]
Good catch @r-zig. I pushed another commit. Want to take another look?

**Comment by mxinden** [comment]
@vmx could you give this pull request a review as a member of the libp2p Rust team?

**Comment by mxinden** [comment]
@vmx thanks for the review. Would you mind taking another look?

**Review by vmx — APPROVED** [review]
Nice! To me that reads really clear now.

**Pull request merged**